### PR TITLE
Snapshot deletion process cleanups

### PR DESCRIPTION
**Pull request by DaveCTurner**
Reorders the methods involved in snapshot deletion to be closer together
and better match the flow of execution, and harmonises the names of many
parameters and local variables to make it easier to follow them through
the process.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
All fairly trivial changes but nonetheless removes quite some obstacles to reading this code already. Lays the foundations for more meaningful follow-ups in this area too.

Presented as a sequence of commits for your reviewing pleasure :)

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -867,33 +829,91 @@ private RepositoryData safeRepositoryData(long repositoryStateId, Map<String, Bl
...
      * @param listener          Listener to invoke once finished
      */
     private void doDeleteShardSnapshots(
         Collection<SnapshotId> snapshotIds,
-        long repositoryStateId,
-        Map<String, BlobContainer> foundIndices,
-        Map<String, BlobMetadata> rootBlobs,
-        RepositoryData repositoryData,
-        IndexVersion repoMetaVersion,
+        long originalRepositoryDataGeneration,
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:905]: Why do we need this parameter here when the other `RepositoryData` parameter already contains this information?

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 2]
```diff
@@ -1117,6 +1127,120 @@ public void onFailure(Exception ex) {
...
+     * as well as any containers for indices that are now completely unreferenced.
+     */
+    private void cleanupUnlinkedRootAndIndicesBlobs(
+        Collection<SnapshotId> snapshotIds,
+        Map<String, BlobContainer> originalIndexContainers,
+        Map<String, BlobMetadata> originalRootBlobs,
+        RepositoryData newRepositoryData,
+        ActionListener<Void> listener
+    ) {
+        cleanupStaleBlobs(snapshotIds, originalIndexContainers, originalRootBlobs, newRepositoryData, listener.map(ignored -> null));
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:1219]: Should we inline this method?

**Review by ywangd — COMMENTED** [review]
LGTM

I don't normally do post-merge reviews. But figured out this one could be useful for my own education. Thanks!

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java (hunk 2 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:1219]: The inlinings are coming :)

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:905]: Eventually this will be gone too I think

**Pull request merged**